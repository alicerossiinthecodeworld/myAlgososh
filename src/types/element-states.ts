export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}

export const colorMap = new Map();
colorMap.set(ElementStates.Default, '4px solid rgb(0, 50, 255)')
colorMap.set(ElementStates.Changing, '4px solid rgb(210, 82, 225)')
colorMap.set(ElementStates.Modified, '4px solid rgb(127, 224, 81)')
