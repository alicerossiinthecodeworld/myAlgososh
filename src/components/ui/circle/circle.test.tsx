import { render, screen, fireEvent } from '@testing-library/react';
import { ElementStates } from '../../../types/element-states';
import { Circle } from './circle';

const testCircleState = (state:ElementStates)=>{
  const { asFragment } = render(<Circle state={state}/>);
  expect(asFragment()).toMatchSnapshot();
}

describe('Circle Component', () => {
  test('renders circle without letter', () => {
    const { asFragment } = render(<Circle/>);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders circle with letter', () => {
    const { asFragment } = render(<Circle letter='a'/>);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders circle with letter as head', () => {

    const { asFragment } = render(<Circle letter='a' head={'head'}/>);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders circle with head', () => {
    const { asFragment } = render(<Circle  head={'head'}/>);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders circle with react element as head', () => {
    const customHead = <div>Custom Head Content</div>;
    const { asFragment } = render(<Circle  head={customHead}/>);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders circle with  tail', () => {
    const { asFragment } = render(<Circle  tail={'tail'}/>);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders circle with react element as tail', () => {
    const customTail = <div>Custom Tail Content</div>;
    const { asFragment } = render(<Circle  head={customTail}/>);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders circle with index', () => {
    const { asFragment } = render(<Circle index={0}/>);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders circle with isSmall', () => {
    const { asFragment } = render(<Circle isSmall={true}/>);
    expect(asFragment()).toMatchSnapshot();
  }); 
  test('renders circle default', () => {
    testCircleState(ElementStates.Default)
  }); 
  test('renders circle changing', () => {
    testCircleState(ElementStates.Changing)
  }); 
  test('renders circle modified', () => {
    testCircleState(ElementStates.Modified)
  }); 
});
