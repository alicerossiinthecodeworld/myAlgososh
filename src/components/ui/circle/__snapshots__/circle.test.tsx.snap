// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Component renders circle changing 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle default 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle modified 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle with  tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      tail
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle with head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      head
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle with index 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      0
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle with isSmall 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle with letter 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        a
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle with letter as head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      head
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        a
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle with react element as head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div>
        Custom Head Content
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle with react element as tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div>
        Custom Tail Content
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component renders circle without letter 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;
